Support for tarball URLs (#2678)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/packages.md

- a tagged release
- a specific commit (full 40-character hash)

Example of a revision specifying a 40-character hash:

```yaml
packages:
  - git: "https://github.com/dbt-labs/dbt-utils.git"
    revision: 4e28d6da126e2940d17f697de783a717f2503188
```

> 

Reviewer: ❤️

As of v0.14.0, dbt will warn you if you install a package using the `git` syntax without specifying a version (see below).

<VersionBlock firstVersion="1.4">

### Internally hosted tarball URL

> Some organizations have security requirements to pull resources only from internal services. To address the need to install packages from hosted environments such as Artifactory or cloud storage buckets, internally hosted tarball URLs are supported for installing packages. 

Reviewer 2: I'd like to clarify who is supporting what and make the sentence active so it will be easier to reduce errors when translated.
```suggestion
Some organizations have security requirements to pull resources only from internal services. To address the need to install packages from hosted environments such as Artifactory or cloud storage buckets, dbt Core enables you to install packages from internally-hosted tarball URLs. 
```

> As the security needs of organizations rapidly evolve, there are increasingly strict requirements to pull resources only from internal services. To address the need to install packages from hosted environments such as Artifactory or cloud storage buckets, internally hosted tarball URLs are supported for installing packages. 

Reviewer: Here's a simplified suggestion: 
```suggestion
Some organizations have security requirements to pull resources only from internal services. To address the need to install packages from hosted environments such as Artifactory or cloud storage buckets, internally hosted tarball URLs are supported for installing packages. 
```

> As the security needs of organizations rapidly evolve, there are increasingly strict requirements to pull resources only from internal services. To address the need to install packages from hosted environments such as Artifactory or cloud storage buckets, internally hosted tarball URLs are supported for installing packages. 

Reviewer 2: In addition to Doug's suggestion:
```suggestion
To help organizations that have tightened security requirements to only pull resources from internal services, dbt supports installing packages from internally-hosted tarball URLs. You can now install packages from hosted environments such as Artifactory or cloud storage buckets. 
```

```yaml
packages:
  - tarball: https://codeload.github.com/dbt-labs/dbt-utils/tar.gz/0.9.6
    name: 'dbt_utils'
```

> Where `name: 'dbt_utils'` specifies the sub-folder of `dbt_packages` that will be created for the package source code to be installed within.

Reviewer 2: love this description! So helpful.
